bundles/automation/org.eclipse.smarthome.automation.rest/src/main/java/org/eclipse/smarthome/automation/rest/internal/RuleResource.java

```diff
@@ -97,16 +97,22 @@ private EnrichedRuleDTO enrich(Rule rule) {
     @Consumes(MediaType.APPLICATION_JSON)
     @ApiOperation(value = "Creates a rule.")
     @ApiResponses(value = { @ApiResponse(code = 201, message = "Created"),
-            @ApiResponse(code = 409, message = "Creation of the rule is refused. Rule with the same UID already exists.") })
+            @ApiResponse(code = 409, message = "Creation of the rule is refused. Rule with the same UID already exists."),
+            @ApiResponse(code = 400, message = "Creation of the rule is refused. Missing required parameter.") })
```

Comment:
We don't have checks for the structure of the json, We are only looking for the required parameters on specified position in the json. In that way the structure is a corner case of missing parameter.